Cookbook/Technical-Documents/Current-guide.md

```diff
@@ -0,0 +1,29 @@
+# `Current` Guide - how to use it in Babylon iOS Project
+
+### Where is that? 
+- `Current` is the global instance of the `World` type which is located in the `BabylonDependencies.framework`. 
+Its purpose is to provide convenient access to the instances that should be shared between different parts of the app (cross-cutting concerns). For more information please read the [How Control the World](/Cookbook/Proposals/ControlTheWorld.md) document.
+
+## `Do`s and `Don't`s
+
+#### Dos
+- access `Current` **only** from `Builder`s layer;
+- pass the required dependencies down the chain;
+- create a customised version to fit your specific needs **for tests**;
+```swift
+let locale = Locale(identifier: "haw_US")
+Current = World(autoupdatingLocale: locale)
```

Comment:
I'm not sure about this one though. Mutating the global `Current` would cause issues in tests if running in parallel - we've seen this issue with snapshots when changing the design library theme in Current and I don't see why the same wouldn't happen when doing it this way

Reply:
(If this is correct, it should be added to the don'ts list)

Reply:
If it can’t be mutated then the 
```#if DEBUG
var Current```
doesn’t make sense anymore. 
It has to be fixed or a `restore` mechanism put in place..
Good point tho 👍

Reply:
Hmmm, you're right. If we make it immutable, then we're essentially back where we were but with normal DI, but without the long initialisers.

Reply:
@Ben-Henshall I think @ilyapuchka 's suggestion here addresses your concerns about the `Current` mutations..
```- reset the instance of `Current` to the default in the `tearDown` method```

Given that, I think we can resolve this now right?

Reply:
Yes, I think it can be one of the reasons we can have instability in tests. I'm more convinced in that than that it is mutating `Current` in the first place - tests that are running in parallel should be running in separate processes thus not sharing a memory, and tests running in one process should be running sequentially, otherwise, I don't know how parallelization works in general 🤷‍♂